### PR TITLE
add windows recommended alerts related to cadvisor and node disk usage alert

### PR DESCRIPTION
**Pull request by Sohamdg081992**
[comment]: # (Note that your PR title should follow the conventional commit format: https://conventionalcommits.org/en/v1.0.0/#summary)
# PR Description
This PR adds the  windows recommended alerts related to cadvisor. Below are the screenshots of the Promql query.

All the other recommended alert queries were tested on our prod cicd [cluster](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/9b96ebbd-c57a-42d1-bbe9-b69296e4c7fb/resourceGroups/ci-prod-aks-mac-weu-rg/providers/Microsoft.ContainerService/managedClusters/ci-prod-aks-mac-weu/overview) and my dev [cluster](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/1a3fd8b1-7a92-4730-8e47-dec9e67f49a9/resourceGroups/testrecalertswcussoham/providers/Microsoft.ContainerService/managedClusters/TestRecAlertsWcusSoham/overview). They are based on kube state metrics and currently works fine on windows os. The only 2 ones are cadvisor based, hence added them in this PR.

The new windows alerts use the similar metrics like the non-windows ones.

nodediskusage

[comment]: # (The below checklist is for PRs adding new features. If a box is not checked, add a reason why it's not needed.)
# New Feature Checklist

- [ ] List telemetry added about the feature.
- [ ] Link to the one-pager about the feature.
- [ ] List any tasks necessary for release (3P docs, AKS RP chart changes, etc.) after merging the PR.
- [ ] Attach results of scale and perf testing.

[comment]: # (The below checklist is for code changes. Not all boxes necessarily need to be checked. Build, doc, and template changes do not need to fill out the checklist.)
# Tests Checklist

- [ ] Have end-to-end Ginkgo tests been run on your cluster and passed? To bootstrap your cluster to run the tests, follow [these instructions](/otelcollector/test/README.md#bootstrap-a-dev-cluster-to-run-ginkgo-tests).
  - Labels used when running the tests on your cluster:
    - [ ] `operator`
    - [ ] `windows`
    - [ ] `arm64`
    - [ ] `arc-extension`
    - [ ] `fips`
- [ ] Have new tests been added? For features, have tests been added for this feature? For fixes, is there a test that could have caught this issue and could validate that the fix works?
  - [ ] Is a new scrape job needed?
    - [ ] The scrape job was added to the folder [test-cluster-yamls](/otelcollector/test/test-cluster-yamls/) in the correct configmap or as a CR. 
  - [ ] Was a new test label added?
    - [ ] A string constant for the label was added to [constants.go](/otelcollector/test/utils/constants.go).
    - [ ] The label and description was added to the [test README](/otelcollector/test/README.md).
    - [ ] The label was added to this [PR checklist](/.github/pull_request_template).
    - [ ] The label was added as needed to [testkube-test-crs.yaml](/otelcollector/test/testkube/testkube-test-crs.yaml).
  - [ ] Are additional API server permissions needed for the new tests?
    - [ ] These permissions have been added to [api-server-permissions.yaml](/otelcollector/test/testkube/api-server-permissions.yaml).
  - [ ] Was a new test suite (a new folder under `/tests`) added?
    - [ ] The new test suite is included in [testkube-test-crs.yaml](/otelcollector/test/testkube/testkube-test-crs.yaml).

### CODE REVIEW

File: .trivyignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 # =========== CRITICAL ================
 CVE-2024-24790
+CVE-2024-5535
```
Comment (gracewehner) [inline .trivyignore:6]: Thanks Soham, do you know which package has this CVE?

Comment (Sohamdg081992) [reply .trivyignore:6]: Thanks Grace! It is from cbl-mariner package and from openssl library.

File: AddonBicepTemplate/recommendedMetricAlerts.bicep [hunk 2]
```diff
@@ -532,6 +554,28 @@ resource recommendedMetricAlertsPodLevel 'Microsoft.AlertsManagement/prometheusR
           }
         ]
       }
+      {
+        alert: 'KubeWindowsContainerAverageCPUHigh'
```
Comment (vishiy) [inline AddonBicepTemplate/recommendedMetricAlerts.bicep:558]: please remove windows & bring them into windows story .

File: AddonBicepTemplate/recommendedMetricAlerts.bicep [hunk 3]
```diff
@@ -320,6 +320,28 @@ resource recommendedMetricAlertsNodeLevel 'Microsoft.AlertsManagement/prometheus
           }
         ]
       }
+      {
+        alert: 'KubeNodeDiskUsageExceeding'
```
Comment (vishiy) [inline AddonBicepTemplate/recommendedMetricAlerts.bicep:324]: rename ?

Comment (Sohamdg081992) [reply AddonBicepTemplate/recommendedMetricAlerts.bicep:324]: Thanks Vishwa!

File: AddonBicepTemplate/recommendedMetricAlerts.bicep (hunk 2 above)
Comment (Sohamdg081992) [reply AddonBicepTemplate/recommendedMetricAlerts.bicep:558]: Sure. Thanks Vishwa!

**Pull request merged**